src/controllers/csi/provisioner/controller.go

```diff
@@ -49,33 +53,62 @@ type OneAgentProvisioner struct {
 	apiReader    client.Reader
 	opts         dtcsi.CSIOptions
 	dtcBuildFunc dynakube.DynatraceClientFunc
-	fs           afero.Fs
 	recorder     record.EventRecorder
 	db           metadata.Access
 	path         metadata.PathResolver
+
+	mu                       *sync.Mutex
+	currentParallelDownloads int64
+	currentlyDownloading     map[string]bool
 }
 
 // NewOneAgentProvisioner returns a new OneAgentProvisioner
-func NewOneAgentProvisioner(mgr manager.Manager, opts dtcsi.CSIOptions, db metadata.Access) *OneAgentProvisioner {
+func NewOneAgentProvisioner(
+	mgr manager.Manager,
+	opts dtcsi.CSIOptions,
+	db metadata.Access) *OneAgentProvisioner {
 	return &OneAgentProvisioner{
-		client:       mgr.GetClient(),
-		apiReader:    mgr.GetAPIReader(),
-		opts:         opts,
-		dtcBuildFunc: dynakube.BuildDynatraceClient,
-		fs:           afero.NewOsFs(),
-		recorder:     mgr.GetEventRecorderFor("OneAgentProvisioner"),
-		db:           db,
-		path:         metadata.PathResolver{RootDir: opts.RootDir},
+		client:               mgr.GetClient(),
+		apiReader:            mgr.GetAPIReader(),
+		opts:                 opts,
+		dtcBuildFunc:         dynakube.BuildDynatraceClient,
+		recorder:             mgr.GetEventRecorderFor(eventRecorderName),
+		db:                   db,
+		path:                 metadata.PathResolver{RootDir: opts.RootDir},
+		mu:                   &sync.Mutex{},
+		currentlyDownloading: map[string]bool{},
```

Comment:
Map's values are not needed here. We need a Set, which can be created this way:
```suggestion
		currentlyDownloading: map[string]interface{},
```

Reply:
empty interface gives us nothing, just longer lines of code

also the correct way is `currentlyDownloading: map[string]interface{}{}` which is ugly as sin 😬

Reply:
we don't use a loop(I delete stuff now), and first result for `golang set` from Google is using a map of bools, so the internet is with me 😛

Reply:
your google search results are profiled up to your needs :stuck_out_tongue:

Reply:
uh, sorry, it's `struct{}`, not `interface{}`!
see: https://www.davidkaya.com/sets-in-golang/

Reply:
that is the 4. result, therefore I still prefer bool 😬